### PR TITLE
Fix colorizer

### LINKED ISSUE #282: Enabling colorizer.sass breaks generally everything.
### ⚠️ Please verify that this bug has NOT been reported before.

- [X] I checked all existing issues and didn't find a similar issue

### Description

Enabling colorizer.sass breaks NvimTree, Fidget, WhichKey, LspSaga, and whatever else depending on which loads before what.

image

### 👟 Reproduction steps

Enable colorizer.sass.

### 👀 Expected behavior

Not this.

### 😓 Actual Behavior

I can't use 90% of my plugins.

### 💻 Metadata

 - system: `"x86_64-linux"`  - host os: `Linux 6.6.29, NixOS, 24.05 (Uakari), 24.05.20240502.63c3a29`  - multi-user?: `yes`  - sandbox: `yes`  - version: `nix-env (Nix) 2.21.2`  - nixpkgs: `/nix/store/p69bcs7ma6ijj8v9xsrg3nq3nn8ryn95-source`

### 📝 Relevant log output

```bash
E5108: Error executing lua .../myNeovimPackages/start/which-key/lua/which-key/view.lua:79: FileType Autocommands for "*": Vim(append):Error executing lua callback: ...eovimPackages/start/nvim-colorizer-lua/lua/colorizer.lua:228: attempt to index field 'sass' (a boolean value)
```

### PR DESCRIPTION
**Pull request by horriblename**
fixes #282 

~~Reverts API changes from the setupOpts PR, since it doesn't work well with this plugin~~

turns out I was looking at the unmaintained repo for a week and wondering why my code doesn't work. `setupOpts` is re-added

### CODE REVIEW

**Comment by horriblename** [comment]
PR is ready but example for `setupOpts.filetypes` relies on #289

File: modules/plugins/ui/colorizer/colorizer.nix [hunk 1]
```diff
@@ -4,101 +4,131 @@
...
+        default = {};
+        example = {
+          "*" = {};
+          "!vim" = {};
+          javascript = {
+            AARRGGBB = false;
+          };
+        };
+        type = submodule {
+          freeformType = attrsOf settingSubmodule;
```
Comment (FrothyMarrow) [inline modules/plugins/ui/colorizer/colorizer.nix:123]: Does this need to be a freeform module, can't it just be:
```nix
type = attrsOf settingSubmoule;
```

**Review by FrothyMarrow — APPROVED** [review]
LGTM

**Review by NotAShelf — APPROVED** [review]
LGTM

**Pull request merged**